examples/tts/conf/hifigan/hifigan_data.yaml

```diff
@@ -0,0 +1,133 @@
+# This config contains the default values for training a HiFi-GAN model.
+# If you want to train model on other dataset, you can change config values according to your dataset.
+# Most dataset-specific arguments are in the head of the config file, see below.
+
+name: "HifiGan"
+
+defaults:
+  - sample: ???
+  - feature: feature_${sample}
+  - model/generator: v1_${sample}
+
+max_epochs: ???
+batch_size: 16
+weighted_sample_steps: null
+
+train_ds_meta: ???
+val_ds_meta: ???
+log_ds_meta: ???
+
+log_dir: ???
+
+model:
+
+  max_epochs: ${max_epochs}
+  steps_per_epoch: ${weighted_sample_steps}
```

Comment:
What does `weighted_sample_steps` mean?

Reply:
Dataloader docstring: `Optional int, If provided, then data will be sampled (with replacement) based on the sample weights provided in the dataset metadata. If None, then sample weights will be ignored.`

In other words, with weighted sampling (and with large datasets in general) the standard definition of an epoch no longer works so you refer to an epoch as a fixed number of training steps.

The name is not as clear as I would like, but alternatives I thought of were either too long (like `weighted_sampling_steps_per_epoch`) or required breaking it into 2 arguments like:

```
WeightedSamplingConfig:
  use_weighted_sampling: true
  steps_per_epoch: 1000
```

Where `steps_per_epoch` does nothing if `use_weighted_sampling: false`.

Reply:
> In other words, with weighted sampling (and with large datasets in general) the standard definition of an epoch no longer works so you refer to an epoch as a fixed number of training steps.

Can't really agree, I always thought of epoch as of something dependent on size of a specific dataset. Number of training examples in the dataset still works, with the caveat that some samples might not have been utilized in a given epoch run.

(Not a blocker)

>  but alternatives I thought of were either too long (like `weighted_sampling_steps_per_epoch`)
I see. True, this is not something that can be expressed consisely. But I see no harm in a long-named argument. I would  even suggest `steps_per_epoch_if_using_weighted_sampling`

Do you thing your could you rename this into a longer and more descriptive version?

Reply:
I tried adding in alternate config which could optionally set `epoch_size = dataset_size` (even if not all samples are seen each epoch), but the way that an epoch is defined for distributed training is relatively complicated (https://github.com/NVIDIA/NeMo/blob/main/nemo/core/optim/lr_scheduler.py#L934-L936). Getting it wrong will create a mismatch with the LR scheduler. So I would rather not support it for now unless there is an urgent need to.